src/vm.rs

```diff
@@ -1182,6 +1195,28 @@ impl<'a, T: GuestStagePagingMode> FinalizedVm<'a, T> {
         )
         .map_err(|_| EcallError::Sbi(SbiError::Failed))?;
 
+        // We need to report guest's htimedelta to Host so that it can schedule timer properly.
+        //
+        //   Salus             Host                TVM
+        //  /                   /                   /
+        // |                   |                   |
+        // |                   |                   |
+        // _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
+        // 0 1 2 3 4 5 6 7 8 9 1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2
+        //                     0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9
+        //                 Time
+        //
+        // For host the TVM's timedelta is
+        //     g_htimedelta = -get_cycle() = -(host time) = -(time + h_htimedelta)
+        //     g_htimedelta = - (20 - 10) = -10
```

Comment:
A possible simplification would be to not use `htimedelta` for the host VM, i.e. host VM time == real time. But this works too, so up to you.

Reply:
That would be problem in migration of the hosts. No ?  
Disclaimer: I still think host in VS mode to actually be useful along with host in HS mode in the future.

Reply:
Migrating the host VM sounds scary and I'm having a hard time thinking of a good use case for it :)

Reply:
I don't think there is any use case right now. May be in the future. However, the cost of maintaining the htimedelta for the host VM is not much. IMO, It follows the standard convention of keeping htimedelta for any VM. Less confusion for future. 
Anyways, I am okay with your suggestion as well. Leave a good comment to explain the reasoning either way.